### PR TITLE
fix: GCC segfault in TrackTests.cpp

### PR DESCRIPTION
**Pull request by paulgessinger**
We've observed the Track unit test executable to SEGFAULT during compilation. 
This seems to be a combination of different things in this compilation unit. I haven't been able to fully figure out if there's a single reason it gets triggered, but in local testing I've found that the problem goes away when I split out this test function into a separate compilation unit. I'm doing that here, and I need to make a small change to our CMake config to make that work.

### CODE REVIEW

**Review by andiwand — APPROVED** [review]

**Comment by andiwand** [comment]
interesting interference

**Comment by paulgessinger** [comment]
I think I have a fix already so let's hold this.

File: Tests/UnitTests/CMakeLists.txt [hunk 1]
```diff
@@ -3,14 +3,16 @@
...
   set(_target "ActsUnitTest${_name}")
-  add_executable(${_target} ${ARGN})
+  add_executable(${_target} ${_source})
   # define required BOOST_TEST_... macros here to ensure consistent names
   target_compile_definitions(
     ${_target}
-    PRIVATE "-DBOOST_TEST_DYN_LINK" "-DBOOST_TEST_MODULE=${_target}")
+    PRIVATE "-DBOOST_TEST_DYN_LINK")
+  set_source_files_properties(${_source} PROPERTIES 
+    COMPILE_DEFINITIONS "BOOST_TEST_MODULE=${_target}")
```
Comment (andiwand) [inline Tests/UnitTests/CMakeLists.txt:15]: what is the change here?

**Review by andiwand — APPROVED** [review]
this is very odd but I guess what counts is that it works

File: Tests/UnitTests/CMakeLists.txt (hunk 1 above)
Comment (paulgessinger) [reply Tests/UnitTests/CMakeLists.txt:15]: `BOOST_TEST_MODULE` is only applied to that single source file now, instead of all source files in the target.

**Pull request merged**